homeassistant/components/enocean/sensor.py

```diff
@@ -3,60 +3,205 @@
 
 import voluptuous as vol
 
+from homeassistant.components import enocean
 from homeassistant.components.sensor import PLATFORM_SCHEMA
-from homeassistant.const import (CONF_NAME, CONF_ID, POWER_WATT)
-from homeassistant.helpers.entity import Entity
+from homeassistant.const import (
+    CONF_DEVICE_CLASS, CONF_ID, CONF_NAME, DEVICE_CLASS_HUMIDITY,
+    DEVICE_CLASS_TEMPERATURE, TEMP_CELSIUS)
 import homeassistant.helpers.config_validation as cv
-from homeassistant.components import enocean
+from homeassistant.helpers.entity import Entity
 
 _LOGGER = logging.getLogger(__name__)
 
+CONF_MAX_TEMP = 'max_temp'
+CONF_MIN_TEMP = 'min_temp'
+CONF_RANGE_FROM = 'range_from'
+CONF_RANGE_TO = 'range_to'
+
 DEFAULT_NAME = 'EnOcean sensor'
 DEPENDENCIES = ['enocean']
 
+DEVICE_CLASS_POWER = 'powersensor'
+
+DOMAIN = 'enocean'
+
+SENSOR_TYPES = {
+    DEVICE_CLASS_HUMIDITY: {
+        'name': 'Humidity',
+        'unit': '%',
+        'icon': 'mdi:water-percent',
+        'class': DEVICE_CLASS_HUMIDITY,
+    },
+    DEVICE_CLASS_POWER: {
+        'name': 'Power',
+        'unit': 'W',
+        'icon': 'mdi:power-plug',
+        'class': DEVICE_CLASS_POWER,
+    },
+    DEVICE_CLASS_TEMPERATURE: {
+        'name': 'Temperature',
+        'unit': TEMP_CELSIUS,
+        'icon': 'mdi:thermometer',
+        'class': DEVICE_CLASS_TEMPERATURE,
+    },
+}
+
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
```

Comment:
Since there will be breaking changes in this PR, I suggest we take the opportunity to move the config sections from the platforms to the component and set up the platforms via discovery from the component. This is our preferred config structure moving forward.

See the netgear_lte component for example config schema.

Reply:
We've agreed to wait with this refactor.